src/main/java/org/folio/linked/data/service/resource/AssignAuthorityTarget.java

```diff
@@ -0,0 +1,43 @@
+package org.folio.linked.data.service.resource;
+
+import static org.apache.commons.collections4.CollectionUtils.isEmpty;
+import static org.folio.ld.dictionary.ResourceTypeDictionary.CONCEPT;
+import static org.folio.ld.dictionary.ResourceTypeDictionary.FAMILY;
+import static org.folio.ld.dictionary.ResourceTypeDictionary.FORM;
+import static org.folio.ld.dictionary.ResourceTypeDictionary.JURISDICTION;
+import static org.folio.ld.dictionary.ResourceTypeDictionary.MEETING;
+import static org.folio.ld.dictionary.ResourceTypeDictionary.ORGANIZATION;
+import static org.folio.ld.dictionary.ResourceTypeDictionary.PERSON;
+import static org.folio.ld.dictionary.ResourceTypeDictionary.PLACE;
+import static org.folio.ld.dictionary.ResourceTypeDictionary.TEMPORAL;
+import static org.folio.ld.dictionary.ResourceTypeDictionary.TOPIC;
+
+import java.util.Collection;
+import java.util.Set;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import org.folio.ld.dictionary.ResourceTypeDictionary;
+
+@Getter
+@RequiredArgsConstructor
+public enum AssignAuthorityTarget {
+
+  CREATOR_OF_WORK(Set.of(PERSON, FAMILY, ORGANIZATION, JURISDICTION, MEETING), Set.of()),
+  SUBJECT_OF_WORK(Set.of(FAMILY, ORGANIZATION, JURISDICTION, MEETING, PERSON, FORM, TOPIC, TEMPORAL, PLACE),
+    Set.of(CONCEPT));
+
+  private final Set<ResourceTypeDictionary> exact;
+  private final Set<ResourceTypeDictionary> any;
+
+  public Boolean isCompatibleWith(Collection<ResourceTypeDictionary> types) {
+    if (isEmpty(types) || types.size() > 2) {
+      return false;
+    }
+    if (types.size() == 1 && exact.containsAll(types)) {
+      return true;
+    }
+    return types.stream().filter(t -> !any.contains(t)).count() == 1
+      && types.stream()
+      .allMatch(t -> exact.contains(t) || any.contains(t));
```

Comment:
This method isn't so readable. 
Could we extract 4 booleans into named private methods to make it clearer please?

Reply:
let me check if we can rewrite it using containsAny

Reply:
done.